neurokit2/data/read_xdf.py

```diff
@@ -53,21 +74,32 @@ def read_xdf(filename, upsample=2, fillmissing=None):
 
       # data, info = nk.read_xdf("data.xdf")
       # sampling_rate = info["sampling_rate"]
+
     """
     try:
         import pyxdf
-    except ImportError:
+    except ImportError as e:
         raise ImportError(
             "The 'pyxdf' module is required for this function to run. ",
             "Please install it first (`pip install pyxdf`).",
-        )
+        ) from e
 
     # Load file
-    # TODO: would be nice to be able to stream a file from URL
+    # if filename is a URL, stream bytes from file
+    if urllib.parse.urlparse(filename).scheme != "":
+        try:
+            req = requests.get(filename, stream=True, timeout=10)
+            req.raise_for_status()  # Raise HTTPError for bad responses (4xx or 5xx)
+
+            req.raw.decode_content = True
+            filename = io.BytesIO(req.content)
+        except requests.exceptions.RequestException as e:
+            raise IOError(f"Failed to read XDF file from URL: {filename}") from e
+
     streams, header = pyxdf.load_xdf(filename)
 
     # Get smaller time stamp to later use as offset (zero point)
-    min_ts = min([min(s["time_stamps"]) for s in streams])
+    min_ts = min(min(s["time_stamps"]) for s in streams)
```

Comment:
I'm not s ure this will work  does it 🤔 
the goal is to  create a vector of mins and then get its min